Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs

```diff
@@ -50,9 +50,9 @@ private void UpdateRanged(float frameTime)
         var bodyQuery = GetEntityQuery<PhysicsComponent>();
         var xformQuery = GetEntityQuery<TransformComponent>();
         var combatQuery = GetEntityQuery<CombatModeComponent>();
-        var query = EntityQueryEnumerator<NPCRangedCombatComponent, TransformComponent>();
+        var query = EntityQueryEnumerator<NPCRangedCombatComponent, TransformComponent, ActiveNPCComponent>();
 
-        while (query.MoveNext(out var uid, out var comp, out var xform))
+        while (query.MoveNext(out var uid, out var comp, out var xform, out _))
         {
             if (comp.Status == CombatStatus.Unspecified)
                 continue;
```

Comment:
Do you know what's not cleaning up? As the comp should be getting dumped when the plan is over.

Reply:
I've had a look into it more thoroughly. It appears at no time does UpdateRanged set a failing CombatStatus for the case of the NPC not being alive. This CombatStatus would be picked up by the RangedOperator and used to decide if the comp would be dumped, as mentioned.

For what it's worth, the Melee system works in this way, not setting a failing status when dead but ostensibly relying on ActiveNPCComponent.

https://github.com/space-wizards/space-station-14/blob/3fa99b8abf41463cf414c2931321e69467c73848/Content.Server/NPC/Systems/NPCCombatSystem.Melee.cs#L96-L107

Reply:
I think I fixed this in the ranged NPC branch as there were a couple of issues. Ideally the mob dying should just dump their plan and in turn dump all of their operators / components.

Can you check if it's still an issue on nyano?

Reply:
Looks like the ranged NPC branch fixed the issue. I'll close this now. Thank you.